src/librustdoc/html/static/css/themes/dark.css

```diff
@@ -304,50 +364,128 @@ a.test-arrow:hover{
-
-pre.should_panic {
-	border-left: 2px solid rgba(255,0,0,.8);
+.tooltip.compile_fail {
+	color: rgba(255,0,0,.8);
 }
-
-pre.should_panic:hover, .information:hover + pre.should_panic {
+.tooltip.compile_fail .icon {
+	/* Used to recolor the icon to the text color. See `.icon` above */
+	filter: url('data:image/svg+xml;utf8, \
+	<svg xmlns="http://www.w3.org/2000/svg"> \
+		<filter id="recolor" color-interpolation-filters="sRGB"> \
+			<feColorMatrix type="matrix" values=" \
+				0 0 0 0 1.0 \
+				0 0 0 0 0.0 \
+				0 0 0 0 0.0 \
+				0 0 0 0.8 0 \
+			"/> \
+		</filter> \
+	</svg> #recolor');
+}
+
+.example-wrap:hover pre.compile_fail {
 	border-left: 2px solid #f00;
 }
-
-pre.ignore {
-	border-left: 2px solid rgba(255,142,0,.6);
-}
-
-pre.ignore:hover, .information:hover + pre.ignore {
-	border-left: 2px solid #ff9200;
+.example-wrap:hover .tooltip.compile_fail {
+	color: #f00;
 }
-
-.tooltip.compile_fail {
-	color: rgba(255,0,0,.8);
+.example-wrap:hover .tooltip.compile_fail .icon {
+	/* Used to recolor the icon to the text color. See `.icon` above */
+	filter: url('data:image/svg+xml;utf8, \
+	<svg xmlns="http://www.w3.org/2000/svg"> \
+		<filter id="recolor" color-interpolation-filters="sRGB"> \
+			<feColorMatrix type="matrix" values=" \
+				0 0 0 0 1.0 \
+				0 0 0 0 0.0 \
+				0 0 0 0 0.0 \
+				0 0 0 1.0 0 \
+			"/> \
+		</filter> \
+	</svg> #recolor');
 }
 
-.information > .compile_fail:hover {
-	color: #f00;
+pre.should_panic {
+	border-left: 2px solid rgba(255,0,0,.8);
 }
-
 .tooltip.should_panic {
 	color: rgba(255,0,0,.8);
 }
-
-.information > .should_panic:hover {
+.tooltip.should_panic .icon {
+	/* Used to recolor the icon to the text color. See `.icon` above */
+	filter: url('data:image/svg+xml;utf8, \
+	<svg xmlns="http://www.w3.org/2000/svg"> \
+		<filter id="recolor" color-interpolation-filters="sRGB"> \
+			<feColorMatrix type="matrix" values=" \
+				0 0 0 0 1.0 \
+				0 0 0 0 0.0 \
+				0 0 0 0 0.0 \
+				0 0 0 0.8 0 \
+			"/> \
+		</filter> \
+	</svg> #recolor');
+}
+
+.example-wrap:hover pre.should_panic {
+	border-left: 2px solid #f00;
+}
+.example-wrap:hover .tooltip.should_panic {
 	color: #f00;
 }
+.example-wrap:hover .tooltip.should_panic .icon {
+	/* Used to recolor the icon to the text color. See `.icon` above */
+	filter: url('data:image/svg+xml;utf8, \
+	<svg xmlns="http://www.w3.org/2000/svg"> \
+		<filter id="recolor" color-interpolation-filters="sRGB"> \
+			<feColorMatrix type="matrix" values=" \
+				0 0 0 0 1.0 \
+				0 0 0 0 0.0 \
+				0 0 0 0 0.0 \
+				0 0 0 1.0 0 \
+			"/> \
+		</filter> \
+	</svg> #recolor');
+}
 
+pre.ignore {
+	border-left: 2px solid rgba(255,142,0,.6);
+}
 .tooltip.ignore {
 	color: rgba(255,142,0,.6);
 }
-
-.information > .ignore:hover {
+.tooltip.ignore .icon {
+	/* Used to recolor the icon to the text color. See `.icon` above */
+	filter: url('data:image/svg+xml;utf8, \
+	<svg xmlns="http://www.w3.org/2000/svg"> \
+		<filter id="recolor" color-interpolation-filters="sRGB"> \
```

Comment:
The support for this feature seems to almost non-existent: https://developer.mozilla.org/en-US/docs/Web/CSS/filter.

I don't think we will be able to use it... Or at least not before a few years.

Reply:
I assume you're referring to the "On SVG elements" row. I don't know what that refers to. I just tested on Chromium 96, and it seems to work fine. Maybe the MDN data is out of date, or maybe that refers to something else. I don't have Safari to test with.

Reply:
We'd need to check what's the oldest browser supporting it. If it's too recent, we'll need to find another way.

Reply:
Regardless, it looks like there may be a way to recolor without the svg url filter. For instance, red can be represented by this transform:
```css
filter: invert(14%) sepia(84%) saturate(7243%) hue-rotate(0deg) brightness(106%) contrast(115%); // 0xf00
```
And then setting the opacity separately with CSS can get us the alpha channel.

For greyscale colors it's even easier, only needing `invert(x%)`

Reply:
I know about this one, I used it a while ago. There is even a website to generate it. Having it as a font has the huge advantage we can rely on simpler styling. Doesn't font-awesome provides the icons inside their font too? Maybe it'd be better to use that instead of svg.

Reply:
There was opposition to pulling in the whole font for just a handful of icons. It may be possible to use something like https://github.com/vusion/webfonts-generator to generate our own font from just the SVGs we use.

Reply:
That would indeed be problematic to add a whole new font for just a few icons. But if we are able to extract only the ones we need (and document how we did in case we need to do it again later!), I think it'd be very acceptable.

Reply:
Okay I'll see what I can get working.